### ISSUE TITLE
auuc_score higher than 1

### ISSUE THREAD

**Issue by s4935324** [uber/causalml#407]
**Describe the bug**
I have used BaseSRegressor and UpliftRandomForestClassifier to compare the performance of feature selection methods, and I am getting auuc score higher than 1. 

**To Reproduce**

          
    #imports
    from causalml.feature_selection.filters import FilterSelect
    from causalml.metrics import plot_gain, auuc_score
    from causalml.inference.tree import UpliftRandomForestClassifier
    from causalml.inference.meta import BaseXRegressor, BaseRRegressor, BaseSRegressor, BaseTRegressor
    from sklearn.model_selection import train_test_split
    from sklearn.ensemble import RandomForestRegressor
    
    #method F
    X_names=data.drop(data[[treatment_col, 'treatment_group_key',outcome_col]],axis = 1).columns
    y_name = outcome_col

    filter_f = FilterSelect()
    
    
    if method == 'KL':
        imp = filter_f.get_importance(data, X_names, y_name, method, 
                          treatment_group = 'treatment1',
                          n_bins=10)
    else:
        imp = filter_f.get_importance(data, X_names, y_name, method, 
                          treatment_group = 'treatment1')
    
    #PERFORMANCE EVALUATION
    # train test split
    df_train, df_test = train_test_split(data, test_size=0.2, random_state=111, stratify = data[y_name])
    
    
    #model
    if model_name==UpliftRandomForestClassifier:
        model = model_name(control_name='control', max_depth=8)
    else :
        model = model_name(RandomForestRegressor(
                                                        n_estimators = 100,
                                                        max_depth = 8,
                                                        min_samples_leaf = 100
                                                        ),control_name='control')

        

    
    # using all features
    features = X_names 
    model.fit(X = df_train[features].values, 
                     treatment = df_train['treatment_group_key'].values,
                     y = df_train[y_name].values)
    y_preds = model.predict(df_test[features].values)
    
    #Select top N features based on KL filter
    top_n = 10
    top_10_features = imp['feature'][:top_n]
    
    top_n = 15
    top_15_features = imp['feature'][:top_n]
    
    top_n = 20
    top_20_features = imp['feature'][:top_n]
    
    #train model again with top features
    treatments = np.where((df_test['treatment_group_key']=='treatment1'), 1, 0)
    
    # using top 10 features
    features = top_10_features 

    model.fit(X = df_train[features].values, 
                     treatment = df_train['treatment_group_key'].values,
                     y = df_train[y_name].values)
    y_preds_t10 = model.predict(df_test[features].values)
    
    # using top 15 features
    features = top_15_features 

    model.fit(X = df_train[features].values, 
                     treatment = df_train['treatment_group_key'].values,
                     y = df_train[y_name].values)
    y_preds_t15 = model.predict(df_test[features].values)
    
    # using top 20 features
    features = top_20_features

    model.fit(X = df_train[features].values, 
                     treatment = df_train['treatment_group_key'].values,
                     y = df_train[y_name].values)
    y_preds_t20 = model.predict(df_test[features].values)
    
    #Print results for Uplift model    
    df_preds = pd.DataFrame([y_preds.ravel(), 
                         y_preds_t10.ravel(),
                         y_preds_t15.ravel(),
                         y_preds_t20.ravel(),
                         treatments,
                         df_test[y_name].ravel()],
                        index=['All', 'Top 10', 'Top 15', 'Top 20', 'is_treated', y_name]).T
    
    
    results = auuc_score(df_preds, outcome_col=y_name, treatment_col='is_treated')

**Expected behavior**
I would expect the score to range between 0 and 1.

**Screenshots**
![image](https://user-images.githubusercontent.com/48221980/138682915-8a003ea9-b371-4151-aaae-3abe6747081f.png)

**Environment (please complete the following information):**
 - Windows
 - Python Version:  3.6
 - Versions of Major Dependencies:
numpy = 1.18.2
pandas = 0.25.3
scipy = 1.4.1
scikit-learn = 0.24.1

**Comment by paullo0106** [comment]
@s4935324 Thanks for flagging this, while I'm still trying to reproduce the issue via our [feature_selection.ipynb](https://github.com/uber/causalml/blob/master/examples/feature_selection.ipynb) example, would be great if you can provide some additional info such as:

1. Is your model_name `UpliftRandomForestClassifier`?  or did the issue happen to both cases
2. Can you help provide a plot by running below code so I can kind of get a sense how the curve look like?
 `plot_gain(df_preds, outcome_col=y_name, treatment_col='is_treated')`

**Comment by s4935324** [comment]
The issue happened to both cases. Here is the plot for Random Forest Classifier:
![image](https://user-images.githubusercontent.com/48221980/142991891-e181ce3d-edaa-440c-a14b-02a27a0bf3d4.png)
And similar pattern I am getting with Regressor:
![image](https://user-images.githubusercontent.com/48221980/142992063-5bd376d3-f14b-4f5c-a86b-0dd094a1eb76.png)

**Comment by jeongyoonlee** [comment]
Hi @s4935324, unlike AUC, AUUC can be greater than 1 because, when the gain in some sub-population is negative, the cumulative gain in the other sub-population (numerator) can be greater than the total gain in the whole population (denominator).

AUUC greater than 1 will not happen if the gain across all population is strictly positive (i.e. no negative treatment effect across population).

Hope this helps.

**Comment by s4935324** [comment]
Thank you @jeongyoonlee for explanation, that is very helpful
